schema.sql

```diff
@@ -13,258 +18,301 @@ drop table if exists areas;
 
+  -- Could overtake some of the fields here.
+  ,operational_cost jsonb null
 
-create table operational_data (
-	unit_id references generation_units(unit_id),
-	fom_cost float not null check (fom_cost >= 0),
-	vom_cost float not null check (vom_cost >=0),
-	forced_outage float not null,
-	startup_cost float not null check (startup_cost >= 0),
-	min_stable_level float not null check (min_stable_level >= 0),
-	mttr integer not null check (mttr >= 0),
-	startup_fuel_mmbtu_per_mw float not null check (startup_fuel_mmbtu_per_mw >= 0),
-	uptime float not null check (uptime >= 0),
-	downtime float not null check (downtime >= 0)
+  -- Enforce unique constraint
+  ,foreign key (entity_id) references entities(id)
 );
 
 -- create table for prime movers
-create table  prime_mover_types (
-	prime_mover text not null,
-	fuel_type text not null,
-	description text null,
-	primary key (prime_mover, fuel_type)
+create table  prime_movers (
+        id integer primary key
+        ,prime_mover text not null
+        ,fuel text null
+        ,description text null
+        ,unique (prime_mover, fuel)
 );
 
 -- create table for balancing topologies
 create table balancing_topologies (
-	name text not null primary key,
-	area text null references areas(name),
-	participation_factor float default 1.0 not null check (participation_factor >= 0 and participation_factor <= 1),
-	description text null
+        id integer primary key
+	,name text not null unique
+	,area text null references areas(name)
+	,participation_factor real default 1.0 not null check (participation_factor >= 0 and participation_factor <= 1)
+	,description text null
 );
 
 -- change to planning regions
 create table  areas (
-	name text not null primary key,
-	description text null
+        id integer primary key
+	,name text not null unique
+	,description text null
 );
 
--- electrical information of the lines
-create table  transmission_lines (
-	balancing_topology_from text references balancing_topologies(name),
-	balancing_topology_to text references balancing_topologies(name),
-	area_from text references areas(name),
-	area_to text references areas(name),
-	continuous_rating float not null check(continuous_rating >= 0),
-	ste_rating float not null check (ste_rating >=0),
-	lte_rating float not null check (lte_rating >=0),
-	line_length float not null check (line_length >= 0)
+-- physical connection between entities
+create table arcs(
+        id integer primary key
+        ,from_to int
+        ,to_from int
+        ,foreign key (from_to) references entities(id)
+        ,foreign key (to_from) references entities(id)
 );
 
+-- transmission lines
+create table transmission_lines (
+        id integer primary key
+        ,arc_id int
+	,continuous_rating real not null check(continuous_rating >= 0)
+	,ste_rating real not null check (ste_rating >=0)
+	,lte_rating real not null check (lte_rating >=0)
+	,line_length real not null check (line_length >= 0)
```

Comment:
Again, what do we do when these are unknown?

Reply:
We are assuming that we have all of this fields.